### PR TITLE
Don't do coverage report until CI done

### PR DESCRIPTION
**Pull request by wxtim**
Experimental PR

**Issue**
Codecov runs before the CI has finished, causing a false negative notification.
[Documentation of setting changed](https://docs.codecov.com/docs/common-recipe-list#send-coverage-notifications-even-if-all-ci-statuses-have-not-completed)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Does not need tests (itself a change to CI setup).
- [x] No change log entry required (developer convienience).
- [x] No documentation update required.

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 # After modifying this file, it might be worth to validate it with:
 # `curl --data-binary @.codecov.yml https://codecov.io/validate`
 
+codecov:
+  notify:
+    require_ci_to_pass: yes
```
Comment (MetRonnie) [inline .codecov.yml:7]: Try?
```suggestion
    require_ci_to_pass: true
```

**Review by MetRonnie — APPROVED** [review]
Should squash the commits down

**Comment by oliver-sanders** [comment]
Looks like it's working.

**Review by oliver-sanders — APPROVED** [review]
Looks fine, didn't get there in time to see if it worked or not, did anyone see it?

**Comment by wxtim** [comment]
> Looks fine, didn't get there in time to see if it worked or not, did anyone see it?

I did, but presumably in an ideal world a reviewer would. I'd be happy to push some silly change, undo and re-push when someone is watching so that it can be checked. Or just to video it then run the vid through a fast forward filter.

**Comment by oliver-sanders** [comment]
Nah, fine, so long as it worked, great.

**Pull request merged**